### PR TITLE
feat(IOBuf): new way to call Get on IOBuf

### PR DESCRIPTION
**Pull request by handong32**
The new Get expects a second parameter,
which is a pointer to some buffer. This
will use memcpy to copy size bytes
into that buffer.

### CODE REVIEW

File: common/src/include/ebbrt/IOBuf.h [hunk 1]
```diff
@@ -169,6 +169,38 @@ class IOBuf {
...
 
+    // copies 'size' bytes into pointer 'vaddr'
+    void GetNoAdvance(size_t size, void* vaddr) {
+      if (!p_)
+        throw std::runtime_error("DataPointer::Get(): past end of buffer");
+
+      assert(p_->Length() > 0);
+
+      // saves vaddr pointer
+      uint8_t* temp = (uint8_t*)vaddr;
```
Comment (dschatzberg) [inline common/src/include/ebbrt/IOBuf.h]: no C style casts please, though this is unnecessary if you change vaddr to be a `uint8_t*`

File: common/src/include/ebbrt/IOBuf.h [hunk 2]
```diff
@@ -179,6 +211,12 @@ class IOBuf {
       return ret;
     }
 
+    // copies 'size' bytes into pointer 'vaddr'
+    void Get(size_t size, void* vaddr) {
```
Comment (dschatzberg) [inline common/src/include/ebbrt/IOBuf.h]: this should really be a `uint8_t *`, and you probably want to call it `buf` or something, not `vaddr`

**Comment by dschatzberg** [comment]
@BetsyRT r+

**Comment by BetsyRT** [comment]
:pushpin: Commit 2c0d905 has been approved by `dschatzberg`

**Comment by BetsyRT** [comment]
:hourglass: Testing commit 2c0d905 with merge 7a3a69c...

**Comment by BetsyRT** [comment]
:broken_heart: Test failed - [runtests](http://192.168.73.2:8010/builders/runtests/builds/45)

**Comment by dschatzberg** [comment]
@BetsyRT r+

**Comment by BetsyRT** [comment]
:pushpin: Commit dc6845b has been approved by `dschatzberg`

**Comment by BetsyRT** [comment]
:hourglass: Testing commit dc6845b with merge 2628710...

**Comment by BetsyRT** [comment]
:sunny: Test successful - [runtests](http://192.168.73.2:8010/builders/runtests/builds/46)
Approved by=dschatzberg
Pushing 262871063a247ba2a12e072045dd9adffedbc8ce to master...

**Pull request merged**